instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java

```diff
@@ -129,17 +126,47 @@ public boolean shouldStart(Context parentContext, REQUEST request) {
    * object of this operation.
    */
   public Context start(Context parentContext, REQUEST request) {
+    return doStart(parentContext, request, null);
+  }
+
+  /**
+   * Ends an instrumented operation. It is of extreme importance for this method to be always called
+   * after {@link #start(Context, Object) start()}. Calling {@code start()} without later {@code
+   * end()} will result in inaccurate or wrong telemetry and context leaks.
+   *
+   * <p>The {@code context} must be the same value that was returned from {@link #start(Context,
+   * Object)}. The {@code request} parameter is the request object of the operation, {@code
+   * response} is the response object of the operation, and {@code error} is an exception that was
+   * thrown by the operation or {@code null} if no error occurred.
+   */
+  public void end(
+      Context context, REQUEST request, @Nullable RESPONSE response, @Nullable Throwable error) {
+    doEnd(context, request, response, error, null);
+  }
+
+  /** Internal method for creating spans with given start/end timestamps. */
+  Context startAndEnd(
+      Context parentContext,
+      REQUEST request,
+      @Nullable RESPONSE response,
+      @Nullable Throwable error,
+      long startTimeNanos,
+      long endTimeNanos) {
```

Comment:
curious your thoughts on:

```suggestion
      Instant startTime,
      Instant endTime) {
```

Reply:
👍